src/quantuminspire/api.py

```diff
@@ -475,7 +475,10 @@ def _create_job(self, name: str, asset: Dict[str, Any], project: Dict[str, Any],
         if not full_state_projection and self.enable_fsp_warning:
             logger.warning("Your experiment can not be optimized and may take longer to execute, "
                            "see https://www.quantum-inspire.com/kbase/optimization-of-simulations/ for details.")
-        return OrderedDict(self._action(['jobs', 'create'], params=payload))
+        try:
+            return OrderedDict(self._action(['jobs', 'create'], params=payload))
+        except (CoreAPIException, TypeError, ValueError) as err_msg:
+            raise ApiError('Job with name {} not created: {}'.format(name, err_msg)) from err_msg
```

Comment:
we try to move to f-strings as much as possible. Could you update these?

Reply:
Did them all in this file